src/mesh/analysis.rs

```diff
@@ -103,6 +103,38 @@ impl BoundingBox {
     pub fn diagonal_length(&self) -> f32 {
         nalgebra::distance(&self.minimum_point, &self.maximum_point)
     }
+
+    /// Check if the two bounding boxes intersect / share any portion of space
+
+    // https://math.stackexchange.com/questions/2651710/simplest-way-to-determine-if-two-3d-boxes-intersect
```

Comment:
Don't change it now, but for the future, feel free to include source links in the doc comments. Also try not having blank lines between doc comments and the things they should be attached to. I ignored the first few cases of you doing this in previous PRs, but I haven't seen this style used anywhere else.

Reply:
autoformatter requires either a blank line starting with `///` or `//` to keep a newline in the doc. without this blank line, If the comment block starts with `///`, autoformatter adds the third slash to the `//` block. that's why I keep the blank line.
I don't mind adding this into the triple slash doc scope though.
![formatter](https://user-images.githubusercontent.com/3333537/71766050-f62c7880-2efb-11ea-828d-d887dbccee4e.gif)

Reply:
Which autoformatter? `rustfmt` doesn't format comments at all, I think. This looks like a VS code specific thing.